sycl/source/detail/scheduler/commands.cpp

```diff
@@ -239,8 +239,20 @@ void Command::waitForEvents(QueueImplPtr Queue,
   if (!EventImpls.empty()) {
     std::vector<RT::PiEvent> RawEvents = getPiEvents(EventImpls);
     if (Queue->is_host()) {
-      const detail::plugin &Plugin = EventImpls[0]->getPlugin();
-      Plugin.call<PiApiKind::piEventsWait>(RawEvents.size(), &RawEvents[0]);
+      // host queue can wait for events from different contexts
```

Comment:
Could you please make this comment a bit more descriptive.
It looks like every context may have it's own plugin, and that caused the need in this change, is that correct understanding?

Reply:
No, this understanding isn't correct.
An example. We have two queues for the same CPU device: `Q1`, `Q2`. Thus we will have two different contexts for the same CPU device: `C1`, `C2`. Also we have default host queue. This queue is accessible via Scheduler. Now, let's assume we have three different events: `E1(C1)`, `E2(C1)`, `E3(C2)`. Also, we have an `EmptyCommand` which is to be executed on host queue. The command's `MPreparedDepsEvents` will contain all three events (`E1`, `E2`, `E3`). Now, if `piEventsWait` is called for all three events we'll experience failure with `CL_INVALID_CONTEXT` 'cause these events refer to different contexts.

OpenCL 2.1 spec says that `clWaitForEvents` will return `CL_INVALID_CONTEXT` if events specified in the list do not belong to the same context.

Now, one may ask how does it happen? Specifically for `EmptyCommand` it's possible due to call to `addDep` from either `addNodeToLeaves` or `updateLeaves`.

Reply:
I'll update the comment.

Reply:
Done